src/logic/RSA.java

```diff
@@ -1,5 +1,63 @@
 package logic;
 
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+
+import javax.crypto.Cipher;
+
 public class RSA {
 
+    private PublicKey publicKey;
+    private PrivateKey privateKey;
+
+    public void generateKeys() throws Exception {
+        KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("RSA");
+        keyPairGenerator.initialize(2048);
+        KeyPair keyPair = keyPairGenerator.generateKeyPair();
+        this.publicKey = keyPair.getPublic();
+        this.privateKey = keyPair.getPrivate();
+    }
+
+    public String encrypt(String plaintext) throws Exception {
+        Cipher cipher = Cipher.getInstance("RSA");
+        cipher.init(Cipher.ENCRYPT_MODE, this.publicKey);
+        byte[] encryptedBytes = cipher.doFinal(plaintext.getBytes());
+        return Base64.getEncoder().encodeToString(encryptedBytes);
+    }
+
+    public String decrypt(String ciphertext, String privateKeyStr) throws Exception {
+        PrivateKey privateKey = loadPrivateKeyFromString(privateKeyStr);
+        Cipher cipher = Cipher.getInstance("RSA");
+        cipher.init(Cipher.DECRYPT_MODE, privateKey);
+        byte[] decryptedBytes = cipher.doFinal(Base64.getDecoder().decode(ciphertext));
+        return new String(decryptedBytes);
+    }
+
+    public String getPrivateKey() {
+        return Base64.getEncoder().encodeToString(this.privateKey.getEncoded());
+    }
+
+    public String getPublicKey() {
+        return Base64.getEncoder().encodeToString(this.publicKey.getEncoded());
+    }
+
+    private PrivateKey loadPrivateKeyFromString(String key) throws Exception {
+        byte[] keyBytes = Base64.getDecoder().decode(key);
+        PKCS8EncodedKeySpec spec = new PKCS8EncodedKeySpec(keyBytes);
+        KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+        return keyFactory.generatePrivate(spec);
+    }
+
+    public void loadPublicKey(String key) throws Exception {
+        byte[] keyBytes = Base64.getDecoder().decode(key);
+        X509EncodedKeySpec spec = new X509EncodedKeySpec(keyBytes);
+        KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+        this.publicKey = keyFactory.generatePublic(spec);
+    }
 }
```

Comment:
You can do the same thing for both of them. I prefer `loadPrivateKeyFromString` will save key to the field like `loadPublicKey`